scripts/consensus_merge.py

```diff
@@ -0,0 +1,587 @@
+
+    for index in (0, 1):
+        normal = 0
+        tumor = 1
+        GT_list = []
+        AD_list = []
+        AF_list = []
+        DP_list = []
+        for caller in CALLER_NAMES:
+            try:
+                targ_var = variant_lookup[caller.lower()]
+            except KeyError:
+                for flist in (GT_list, AD_list, AF_list, DP_list):
+                    flist.append('.')
+                continue
+            if index == normal:
+                targ_sample = targ_var.normal
+            elif index == tumor:
+                targ_sample = targ_var.tumor
+                
+            GT_list.append(str(targ_sample.GT))
+            AD_list.append(stringify(targ_sample.AD))
+            AF_list.append('{:0.4f}'.format(targ_sample.AF))
+            DP_list.append(str(targ_sample.DP))
+
+        GT = stringify(GT_list, FORMAT_JOIN)
+        AD = stringify(AD_list, FORMAT_JOIN)
+        AF = stringify(AF_list, FORMAT_JOIN)
+        DP = stringify(DP_list, FORMAT_JOIN)
+
+        output_record.samples[index]['GTC'] = GT
+        output_record.samples[index]['ADC'] = AD
+        output_record.samples[index]['AFC'] = AF
+        output_record.samples[index]['DPC'] = DP
+
+        consensus_gt, gt_tag = get_gt_consensus(GT_list)
+
+        output_record.samples[index]['AGT'] = consensus_gt
+        output_record.samples[index]['GT_STATUS'] = gt_tag
+    
+    chrom = single_caller_variants[0].record.chrom
+    pos = single_caller_variants[0].record.pos
+    mapq, mq0 = get_mapq(normal_cram, chrom, pos)
+
+    output_record.info['MQ'] = mapq
+    output_record.info['MQ0'] = mq0
+    output_record.info['CAL'] = ','.join([c for c in CALLER_NAMES if c.lower() in variant_lookup])
+
+    if hotspot:
+        output_record.filter.add('Hotspot')
+    else:
+        output_record.filter.add('Consensus')
+
+    output_vcf.write(output_record)
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--strelka2_vcf', 
+                        help='Strelka2 VCF with MNPs fixed')
+    parser.add_argument('--mutect2_vcf',
+                        help='Mutect2 VCF')
+    parser.add_argument('--lancet_vcf', 
+                        help='Lancet VCF')
+    parser.add_argument('--vardict_vcf',
+                        help='VarDict VCF')
+    parser.add_argument('--cram', help='CRAM file')
+    parser.add_argument('--output_basename',
+                        help='String to use as basename for output file')
+
+    args = parser.parse_args()
+
+    # Create VCF and CRAM pysam objects
+    strelka2_vcf = pysam.VariantFile(args.strelka2_vcf, 'r')
+    mutect2_vcf = pysam.VariantFile(args.mutect2_vcf, 'r')
+    lancet_vcf = pysam.VariantFile(args.lancet_vcf, 'r')
+    vardict_vcf = pysam.VariantFile(args.vardict_vcf, 'r')
+
+    normal_cram = pysam.AlignmentFile(args.cram, 'rc', reference_filename="/home/ubuntu/volume/ref/Homo_sapiens_assembly38.fasta.fai")
```

Comment:
I'd make the ref file name an arg

Reply:
The reference shouldn't actually be necessary at all. I was unable to get the random access to work without providing it, and I was thinking there was something wrong with my pysam or with the environment on my instance, but actually it looks like there's something wrong with the cram file I downloaded from Cavatica. Everything runs along smoothly until it gets to chr11 and then it dies for some reason. I will comment the ref out of the version I push and hopefully no one else has this problem.

Reply:
Um, hmmm, how did you download the file? It could be that the cram just didn't download completely. Also, the `reference_filename` should be `Homo_sapiens_assembly38.fasta`, you used the index instead. I think it needs to be indexed to work, but you should give just the fasta file name as the input.

Reply:
I got the 'download link' in Cavatica and used wget. The cram isn't just truncated because it works all the way to the end as long as you provide that local reference. I tried re-downloading it but the problem persisted. Super weird about using the index instead of the fasta! I don't think I did that on purpose, but it appears to work using either, which I do not understand. Pointing to nonexistent files or other local files that are neither fastas nor their indices doesn't work, though.

Reply:
Hmmm, this may be a dumb question, but did you also download the cram index? Just trying to think if what I may have done different as the counts from cram script that I wrote a while back did not throw errors...

Reply:
I did not actually see the index at the [location](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/pbta-lancet-vardict-analysis/tasks/2170f29d-c095-4c50-8d43-91a96550fac1/) from which I got the cram and could not otherwise find it, but I then generated it using the 'samtools index' app on Cavatica. I think I wasn't able to generate it with samtools locally, which is probably a clue.